testsuite/tests/typing-jkind-bounds/subsumption/quality.ml

```diff
@@ -240,33 +237,11 @@ end = struct
   type 'a t constraint 'a = [< `a of string | `b]
 end
 [%%expect {|
-Lines 4-7, characters 6-3:
-4 | ......struct
-5 |   type 'a u = [< `a of string | `b] as 'a
-6 |   type 'a t constraint 'a = [< `a of string | `b]
-7 | end
-Error: Signature mismatch:
-       Modules do not match:
-         sig
-           type 'a u = 'a constraint 'a = [< `a of string | `b ]
-           type 'a t constraint 'a = [< `a of string | `b ]
-         end
-       is not included in
-         sig
-           type 'a u = 'a constraint 'a = [< `a of string | `b ]
-           type 'a t : value mod global with [< `a of string | `b ] u
-             constraint 'a = [< `a of string | `b ]
-         end
-       Type declarations do not match:
-         type 'a t constraint 'a = [< `a of string | `b ]
-       is not included in
-         type 'a t : value mod global with [< `a of string | `b ] u
-           constraint 'a = [< `a of string | `b ]
-       The kind of the first is value
-         because of the definition of t at line 6, characters 2-49.
-       But the kind of the first must be a subkind of value mod global
-         with [< `a of string | `b ] u
-         because of the definition of t at line 3, characters 2-40.
+module M :
+  sig
+    type 'a u = 'a constraint 'a = [< `a of string | `b ]
+    type 'a t constraint 'a = [< `a of string | `b ]
```

Comment:
This one is ok.... but I'm worried we have a bestness problem here. Set up the same test with portability and replace the `string` with a function. I'd want to make sure that we just expand `'a u` to `'a` and not make any assumptions about `u`'s kind.

Reply:
Added a test, but not sure that I got it right.

Reply:
Left some `CR reisenberg`s around this (on call)